lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -317,6 +337,127 @@ fn parse_dogstatsd<'a>(
+                event::TIMESTAMP_PREFIX => {
+                    let (_, timestamp) = all_consuming(preceded(tag(event::TIMESTAMP_PREFIX), unix_timestamp))(chunk)?;
+                    maybe_timestamp = Some(timestamp);
+                }
+                // Hostname: client-provided hostname for the host that this event originated from.
+                event::HOSTNAME_PREFIX => {
+                    let (_, hostname) =
+                        all_consuming(preceded(tag(event::HOSTNAME_PREFIX), event_option_value))(chunk)?;
+                    maybe_hostname = Some(hostname.to_string());
+                }
+                // Aggregation key: key to be used to group this event with others that have the same key.
+                event::AGGREGATION_KEY_PREFIX => {
+                    let (_, aggregation_key) =
+                        all_consuming(preceded(tag(event::AGGREGATION_KEY_PREFIX), event_option_value))(chunk)?;
+                    maybe_aggregation_key = Some(aggregation_key.to_string());
+                }
+                // Priority: client-provided priority of the event.
+                event::PRIORITY_PREFIX => {
+                    let (_, priority) =
+                        all_consuming(preceded(tag(event::PRIORITY_PREFIX), event_option_value))(chunk)?;
+                    match priority {
+                        event::PRIORITY_LOW => maybe_priority = Some(saluki_event::eventd::Priority::Low),
+                        _ => maybe_priority = Some(saluki_event::eventd::Priority::Normal),
+                    }
+                }
+                // Source type name: client-provided source type name of the event.
+                event::SOURCE_TYPE_PREFIX => {
+                    let (_, source_type) =
+                        all_consuming(preceded(tag(event::SOURCE_TYPE_PREFIX), event_option_value))(chunk)?;
+                    maybe_source_type = Some(source_type.to_string());
+                }
+                // Alert type: client-provided alert type of the event.
+                event::ALERT_TYPE_PREFIX => {
+                    let (_, alert_type) =
+                        all_consuming(preceded(tag(event::ALERT_TYPE_PREFIX), event_option_value))(chunk)?;
+                    match alert_type {
+                        event::ALERT_TYPE_ERROR => maybe_alert_type = Some(saluki_event::eventd::AlertType::Error),
+                        event::ALERT_TYPE_WARNING => maybe_alert_type = Some(saluki_event::eventd::AlertType::Warning),
+                        event::ALERT_TYPE_SUCCESS => maybe_alert_type = Some(saluki_event::eventd::AlertType::Success),
+                        _ => maybe_alert_type = Some(saluki_event::eventd::AlertType::Info),
+                    }
+                }
+                // Tags: additional tags to be added to the metric.
+                event::TAGS_PREFIX => {
+                    let (_, tags) = all_consuming(preceded(tag(event::TAGS_PREFIX), metric_tags(config)))(chunk)?;
```

Comment:
While it's probably not a bad idea to follow the same logic as metrics, it would be good to check, and maybe even add a link to the code, showing that tags in events follow the same restrictions as tags in metrics.

Reply:
Yea, will need to look to see if this is documented somewhere